runtime/lua/vim/lsp/rpc.lua

```diff
@@ -608,6 +608,73 @@ function M.connect(host, port)
   end
 end
 
+---@class Dispatchers
+---@field notification function
+---@field server_request function
+---@field on_exit function
+---@field on_error function
+
+-- TODO: better name
+---@class StartServer
+---@field request fun(method: string, params: table?, callbackfun: fun(err: lsp.ResponseError | nil, result: any), notify_reply_callback:function?)
+---@field notify fun(method: string, params: any)
+---@field is_closing function
+---@field terminate function
+
+--- Create a LSP RPC client factory that connects via named pipes (window)
+--- or unix domain sockets (unix) to the given pipe_path (file path on
+--- Unix and name on Windows)
+---
+---@param pipe_path  string
+---@return fun(dispatchers: Dispatchers): StartServer?
```

Comment:
Since this is part of the public API, consider adding descriptions to the parameters/return values.